### ISSUE TITLE
EmbeddedFileProvider: Not recognizing correct manifest resource names

### ISSUE THREAD

**Issue by Yannici** [dotnet/aspnetcore#45585]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Describe the bug

When using the EmbeddedFileProvider in my example for the UseStaticFiles builder method, it won't recognizing that the manifest resource names of the folders with an hypen (-) changes to underscore (_). So it cannot find the embedded files.

In my researches I found out, that the hyphen may not be an allowed EverettId character and gets replaced by an underscore in the [CreateManifestNameImpl](https://github.com/dotnet/msbuild/blob/main/src/Tasks/CreateCSharpManifestResourceName.cs#L89)-Method of the [CreateCSharpManifestResourceName.cs](https://github.com/dotnet/msbuild/blob/main/src/Tasks/CreateCSharpManifestResourceName) class.

**Example:**

File path: 
/lib/nice-css/nice-css.css

Manifest resource name: 
lib.nice_css.nice-css.css

Embedded File Provider looking for path:
lib.nice-css.nice-css.css

### Expected Behavior

The embedded file provider should looking for the correct resource name with the same logic how the resource names gets created.

### Steps To Reproduce

https://github.com/Yannici/IssueEmbeddedFileProvider

### .NET Version

7.0.100

### Anything else?

Visual Studio 2022 17.4.2
ASP.NET Core Version: 7.0.0

.NET SDK:
 Version:   7.0.100
 Commit:    e12b7af219

Laufzeitumgebung:
 OS Name:     Windows
 OS Version:  10.0.19042
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\7.0.100\

Host:
  Version:      7.0.0
  Architecture: x64
  Commit:       d099f075e4

.NET SDKs installed:
  7.0.100 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 3.1.31 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.31 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.11 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.31 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.11 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.0 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
  Not set

global.json file:
  Not found

**Comment by javiercn** [comment]
@Yannici thanks for contacting us.

Feel free to send us a PR if you are interested in providing a fix.

**Comment by Yannici** [comment]
> @Yannici thanks for contacting us.
> 
> Feel free to send us a PR if you are interested in providing a fix.

Thanks @javiercn for the response.
I would like to do an PR, but I need some assistance to set up my code.

I want to use the publich static method [MakeValidEverettIdentifier](https://github.com/dotnet/msbuild/blob/a6f6699d1f70bf79db82030938d2c5e52d1e4d2e/src/Tasks/CreateManifestResourceName.cs#L372). But this is in the msbuild repository, and I don't know if I can reference this repo/assembly in any way?

If this is not possible, I think about copy the relevant methods, but would be just bad practice I guess...

**Comment by Yannici** [comment]
I made a commit in my fork to fix this issue. I added a dependency to `Microsoft.Build.Tasks.Core`. Problem is, that it's losing backwards compatibility with net462 because of this referenced assembly.
Maybe someone can help me here? Is there a possibility to keep the backwards compatibility and use the existing code from the `Microsoft.Build.Tasks` library?

Here is the commit:
https://github.com/dotnet/aspnetcore/commit/b53170b2aa7d42041ea436dd10980068a07b11c7
